contracts/tests/mcms/ManyChainMultiSigSubgroups.spec.ts

```diff
@@ -0,0 +1,528 @@
+
+    // Set configuration
+    {
+      const signers = asSnakeData<bigint>(
+        testSigners.map((s) => BigInt('0x' + s.keyPair.publicKey.toString('hex'))),
+        (a) => beginCell().storeUint(a, 256),
+      )
+
+      const setConfigBody = mcms.builder.message.in.setConfig.encode({
+        queryId: 1n,
+        signerKeys: signers,
+        signerGroups: asSnakeData<number>(
+          testSigners.map((s) => s.group),
+          (g) => beginCell().storeUint(g, 8),
+        ),
+        groupQuorums,
+        groupParents,
+        clearRoot: false,
+      })
+
+      const result = await bind.mcms.sendInternal(
+        acc.multisigOwner.getSender(),
+        toNano('1'),
+        setConfigBody,
+      )
+      expect(result.transactions).toHaveTransaction({
+        from: acc.multisigOwner.address,
+        to: bind.mcms.address,
+        success: true,
+      })
+    }
+
+    const [nonce, newState] = await randomBetween(randomState, 0, 2 ** 32)
+    randomState = newState
+    const testOps: mcms.Op[] = [
+      {
+        chainId: MCMSBaseTestSetup.TEST_CHAIN_ID,
+        multiSig: bind.mcms.address,
+        nonce: BigInt(nonce),
+        to: bind.mcms.address, // TODO bind.counter.address,
+        value: toNano('0.1'),
+        data: beginCell().storeUint(0xffffffff, 32).endCell(),
+      },
+    ]
+    // Test setRoot with non-override metadata
+    const rootMetadata: mcms.RootMetadata = {
+      chainId: MCMSBaseTestSetup.TEST_CHAIN_ID,
+      multiSig: bind.mcms.address,
+      preOpCount: 0n,
+      postOpCount: 1n,
+      overridePreviousRoot: false,
+    }
+
+    const signers = testSigners.map((s) => ({
+      publicKey: s.keyPair.publicKey,
+      sign: (data: Buffer) => sign(data, s.keyPair.secretKey),
+    }))
+
+    if (!allSignersNeeded) {
+      // Can remove at least one signature and setRoot still works
+      let success = true
+      for (let i = 0; i < signers.length; i++) {
+        const reducedSigners = removeIndex(signers, i)
+        const [nonce, newState] = await randomBetween(randomState, 0, 2 ** 32)
+        randomState = newState
+        const testOps: mcms.Op[] = [
+          {
+            chainId: MCMSBaseTestSetup.TEST_CHAIN_ID,
+            multiSig: bind.mcms.address,
+            nonce: BigInt(nonce),
+            to: bind.mcms.address, // TODO bind.counter.address,
+            value: toNano('0.1'),
+            data: beginCell().storeUint(0xffffffff, 32).endCell(),
+          },
+        ]
+        const [reducedSetRoot, opProofs] = merkleProof.build(
+          reducedSigners,
+          BigInt(MCMSBaseTestSetup.TEST_VALID_UNTIL),
+          rootMetadata,
+          testOps,
+        )
+
+        const reducedSetRootBody = mcms.builder.message.in.setRoot.encode(reducedSetRoot)
+        const result = await bind.mcms.sendInternal(
+          acc.deployer.getSender(),
+          toNano('1'),
+          reducedSetRootBody,
+        )
+
+        const transaction = result.transactions.find(
```

Comment:
Why can't we check this with the usual `expect(r.transactions).toHaveTransaction`?